soroban-env-common/src/raw_val.rs

```diff
@@ -59,6 +59,18 @@ pub enum Tag {
     Reserved = 7,
 }
 
+impl Tag {
+    pub const fn rawval_const(&self) -> i64 {
+        ((*self as i64) << 1) | 1
+    }
+    pub const fn rawval_mask() -> i64 {
+        ((TAG_MASK as i64) << 1) | 1
+    }
+}
+
+sa::const_assert!(Tag::I32.rawval_const() == 3);
+sa::const_assert!(Tag::rawval_mask() == 0xf);
```

Comment:
What are these for?

Reply:
They're used in `assert_val_tag` just to check that a given RawVal has a given tag (simultaneously checking that it has the 1-bit set). We could use the existing constants but they don't include the low 1-bit, they're for checking the tag separately from the 1-bit, as done in the RawVal code.